app/src/main/java/com/gh4a/activities/issues_label/IssueLabelListActivity.java

```diff
@@ -284,129 +274,30 @@ public void onDestroyActionMode(ActionMode mode) {
             mAdapter.notifyDataSetChanged();
             updateFabVisibility();
         }
-    }
-
-    private class DeleteIssueLabelTask extends ProgressDialogTask<Void> {
-        private final String mLabelName;
-
-        public DeleteIssueLabelTask(String labelName) {
-            super(IssueLabelListActivity.this, R.string.deleting_msg);
-            mLabelName = labelName;
-        }
-
-        @Override
-        protected ProgressDialogTask<Void> clone() {
-            return new DeleteIssueLabelTask(mLabelName);
-        }
-
-        @Override
-        protected Void run() throws IOException {
-            LabelService labelService = (LabelService)
-                    Gh4Application.get().getService(Gh4Application.LABEL_SERVICE);
-            labelService.deleteLabel(mRepoOwner, mRepoName, URLEncoder.encode(mLabelName, "UTF-8"));
-            return null;
-        }
 
-        @Override
-        protected void onSuccess(Void result) {
-            forceLoaderReload(0);
-            setResult(RESULT_OK);
-        }
-
-        @Override
-        protected String getErrorMessage() {
-            return getContext().getString(R.string.issue_error_delete_label, mLabelName);
-        }
-    }
-
-    private class EditIssueLabelTask extends ProgressDialogTask<Void> {
-        private final String mOldLabelName;
-        private final String mNewLabelName;
-        private final String mColor;
-
-        public EditIssueLabelTask(String oldLabelName, String newLabelName, String color) {
-            super(IssueLabelListActivity.this, R.string.saving_msg);
-            mOldLabelName = oldLabelName;
-            mNewLabelName = newLabelName;
-            mColor = color;
-        }
-
-        @Override
-        protected ProgressDialogTask<Void> clone() {
-            return new EditIssueLabelTask(mOldLabelName, mNewLabelName, mColor);
+        public void addIssue() {
+            IssuesLabelService.addIssue(IssueLabelListActivity.this, getRootLayout(), mRepoOwner, mRepoName, mLabel.editedName, mLabel.editedColor)
+                .doOnError(error -> {
+                    mAdapter.remove(mAddedLabel);
+                    mAddedLabel = null;
+                })
+                .subscribe(result -> {
+                    forceLoaderReload(0);
+                    mAddedLabel = null;
+                    setResult(RESULT_OK);
+                }, e -> {});
```

Comment:
Use this error handler instead of `doOnError` above?

Reply:
I'm sorry, I did not understand, could you detail that pls ?

Reply:
With this I meant the empty error handler:
`e -> {}`